app/earn/components/vault-card/vault-user-balance.tsx

```diff
@@ -0,0 +1,64 @@
+import { Spacer, Spinner, Tooltip } from '@nextui-org/react'
+import type { Vault } from '../../../api'
+import CompactNumber from '../../../components/compact-number/compact-number'
+import { useAppSelector } from '../../../store/hooks'
+import { toStringNumberFromDecimals, toUSDValue } from '../../../shared'
+import { Row } from '../../../components/row/Row'
+import { useWalletWrapperContext } from '../../../providers/wallet/wallet-wrapper-provider'
+import { WalletStatus } from '../../../providers/wallet/wrappers/types'
+
+interface Props {
+  vault: Vault
+}
+
+export function VaultUserBalance({ vault }: Props) {
+  const { status: walletStatus } = useWalletWrapperContext()
+  const { vaultBalances, isLoading } = useAppSelector(state => state.positionInfo)
+  const { symbol: assetSymbol, decimals } = vault.asset
+  const balance = vaultBalances[vault.address] ?? 0n
+  const { value: price, decimals: priceDecimals } = vault.asset.price
+  const threshold = BigInt(1e6) * 10n ** BigInt(decimals)
+
+  return (
+    (walletStatus === WalletStatus.CONNECTING || isLoading) ? <Loader /> : <Value />
+  )
+
+  function Value() {
+    if (walletStatus === WalletStatus.NOT_CONNECTED) {
+      return '-'
```

Comment:
In such case better not display anything at all

Reply:
ok